src/DbHandler.php

```diff
@@ -0,0 +1,208 @@
+     * @param int       $maxEntries
+     * @param array     $additionalFields
+     * @param int       $level
+     * @param bool      $bubble
+     * @param string    $table
+     */
+    public function __construct(
+        Db $db,
+        $maxEntries = null,
+        array $additionalFields = [],
+        $level = Logger::DEBUG,
+        $bubble = true,
+        $table = 'log'
+    ) {
+        parent::__construct($level, $bubble);
+
+        $this->db = $db;
+        $this->maxEntries = $maxEntries;
+        $this->additionalFields = $additionalFields;
+        $this->table = $table;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function processRecord(array $record)
+    {
+        $record = parent::processRecord($record);
+
+        $record['additionalFieldsData'] = [];
+        foreach ($this->additionalFields as $field) {
+            foreach (['context', 'extra'] as $sourceKey) {
+                if (isset($record[$sourceKey]) && array_key_exists($field, $record[$sourceKey])) {
+                    $record['additionalFieldsData'][$field] = $record[$sourceKey][$field];
+                    unset($record[$sourceKey][$field]);
+                }
+            }
+        }
+
+        return $record;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function write(array $record)
+    {
+        $dbData = [
+            'channel' => $record['channel'],
+            'level'   => $record['level_name'],
+            'message' => $record['formatted'],
+        ] + $record['additionalFieldsData'];
+
+        $this->db->insert($this->table, $dbData);
+
+        $this->clean($record['channel']);
+    }
+
+    /**
+     * @param string $channel
+     */
+    protected function clean($channel)
+    {
+        if ($this->maxEntries && mt_rand(1, $this->cleanDivisor) <= $this->cleanProbability) {
+            $currentCount = $this->getChannelEntriesCount($channel);
+            if ($currentCount > $this->maxEntries) {
+                $entriesToDelete = $currentCount - $this->maxEntries;
+                $this->deleteXOldestChannelEntries($channel, $entriesToDelete);
+            }
+        }
+    }
+
+    /**
+     * @param string $channel
+     * @return int
+     */
+    protected function getChannelEntriesCount($channel)
+    {
+        return $this->db->fetchValue(
+            'SELECT COUNT(*) FROM `' . $this->table . '` WHERE `channel` = ?',
+            [$channel]
+        );
+    }
+
+    /**
+     * @param $channel
+     * @param $entriesToDelete
+     */
+    protected function deleteXOldestChannelEntries($channel, $entriesToDelete)
+    {
+        $this->db->exec(
+            sprintf(
+                'DELETE FROM `%s` WHERE `channel` = ?  ORDER BY `time` ASC LIMIT %d',
+                $this->table,
+                (int) $entriesToDelete
+            ),
+            [$channel]
+        );
+    }
+
+    /**
+     * @param Logger $logger
+     */
+    public function clear(Logger $logger)
```

Comment:
```suggestion
    public function clear(Logger $logger): void
```
And the PHPDocs can be removed as well (here and elsewhere)